Reply/app/src/main/java/com/materialstudies/reply/ui/compose/ComposeFragment.kt

```diff
@@ -137,92 +124,76 @@ class ComposeFragment : Fragment() {
-
-            // Since View to View transforms often are not transforming into full screens,
-            // remove the transition's scrim.
             scrimColor = Color.TRANSPARENT
+            // Have the transform match the endView card's native elevation as closely as possible.
+            endElevation = requireContext().resources.getDimension(
+                R.dimen.email_recipient_card_popup_elevation_compat
+            )
+            // Avoid having this transform from running on both the start and end views by setting
+            // its target to the endView.
+            addTarget(binding.recipientCardView)
+        }
 
+        // Configure the analogous collapse transform back to the recipient chip. This should
+        // happen when the card is clicked, any region outside of the card (the card's transparent
+        // scrim) is clicked, or when the back button is pressed.
+        binding.run {
+            recipientCardView.setOnClickListener { collapseChip(chip) }
+            recipientCardScrim.setOnClickListener { collapseChip(chip) }
         }
-        // Begin the transition by changing properties on the start and end views or
-        // removing/adding them from the hierarchy.
+        closeRecipientCardOnBackPressed.expandedChip = chip
+        closeRecipientCardOnBackPressed.isEnabled = true
+
         TransitionManager.beginDelayedTransition(binding.composeConstraintLayout, transform)
         binding.recipientCardView.visibility = View.VISIBLE
-        binding.recipientChipGroup.visibility = View.GONE
-
+        // Using INVISIBLE instead of GONE ensures the chip's parent layout won't shift during
+        // the transition due to chips being effectively removed.
+        chip.visibility = View.INVISIBLE
     }
 
-    private fun containerTransformRecipientCardToChip() {
+    /**
+     * Collapse the recipient card back into its [chip] form.
+     */
+    private fun collapseChip(chip: View) {
         val transform = MaterialContainerTransform().apply{
-            // Manually tell the container transform which Views to transform between.
             startView = binding.recipientCardView
-            endView = binding.recipientChipGroup
-
-            // Optionally add a curved path to the transform
-            // pathMotion = MaterialArcMotion()
-
-            // Since View to View transforms often are not transforming into full screens,
-            // remove the transition's scrim.
+            endView = chip
             scrimColor = Color.TRANSPARENT
+            startElevation = requireContext().resources.getDimension(
+                R.dimen.email_recipient_card_popup_elevation_compat
+            )
+            addTarget(chip)
+        }
 
+        // Clear the scrim listener and on back pressed callbacks
+        binding.recipientCardScrim.run {
+            setOnClickListener {  }
+            isClickable = false
+            isFocusable = false
         }
```

Comment:
i think it would be clearer to have the scrim be always clickable in XML, but GONE initially, and then we show/hide it during expand/collapse.

that way it isn't in the view hierarchy at all when it's not supposed to be, and i think it's a more realistic setup, e.g. if we gave it a dim color with opacity you wouldn't be able to just change clickable-ness here.

Reply:
Agreed! Done